vdp/pipeline/v1beta/pipeline.proto

```diff
@@ -44,34 +44,147 @@ message ReadinessResponse {
   common.healthcheck.v1beta.HealthCheckResponse health_check_response = 1;
 }
 
-// Component is the fundamental building block in pipelines.
+// StartComponent
+// Configures the starting point for pipeline triggering.
+message StartComponent {
+  // Represents a field within the start component.
+  message Field {
+    // Title of the field.
+    string title = 1 [(google.api.field_behavior) = OPTIONAL];
+    // Description of the field.
+    string description = 2 [(google.api.field_behavior) = OPTIONAL];
+    // Instill format.
+    string instill_format = 3 [(google.api.field_behavior) = OPTIONAL];
+    // Instill UI order.
+    int32 instill_ui_order = 4 [(google.api.field_behavior) = OPTIONAL];
+  }
+  // Fields configuration.
+  // Key: Key of the input data.
+  // Field: Field settings of the value.
+  map<string, Field> fields = 1 [(google.api.field_behavior) = OPTIONAL];
+}
+
+// EndComponent
+// Configures the ending point for pipeline triggering.
+message EndComponent {
+  // Represents a field within the end component.
+  message Field {
+    // Title of the field.
+    string title = 1 [(google.api.field_behavior) = OPTIONAL];
+    // Description of the field.
+    string description = 2 [(google.api.field_behavior) = OPTIONAL];
+    // Value of the field.
+    string value = 3 [(google.api.field_behavior) = OPTIONAL];
+    // Instill UI order.
+    int32 instill_ui_order = 4 [(google.api.field_behavior) = OPTIONAL];
+  }
+  // Fields configuration.
+  // Key: Key of the output data.
+  // Field: Field settings of the value.
+  map<string, Field> fields = 1 [(google.api.field_behavior) = OPTIONAL];
+}
+
+// ConnectorComponent
+// Configures a connector component. Requires the creation of a connector resource first.
+message ConnectorComponent {
+  // Definition name.
+  string definition_name = 1 [(google.api.field_behavior) = OPTIONAL];
+  // Connector definition.
+  ConnectorDefinition definition = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Resource name. Leave empty if not created.
+  string resource_name = 3 [(google.api.field_behavior) = OPTIONAL];
+  // Describes the connector resource details (e.g., configuration to connect with the vendor service).
+  Connector resource = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
```

Comment:
This is one of the points that might be unclear. The way I understood the entitiy names, operator/connector/iterator are components _within a pipeline_. We've called them here `ConnectorComponent`, `OperatorComponent`, etc.

We have this lingering `Connector` type, which is particular to this type of component, and that refers to the resource a user needs to create in order to use a connector component.

I think it would be clearer if we renamed this as `ConnectorResource`, I don't know if precisely because we're introducing breaking changes we can afford doing this extra refactor or if on the other hand we want to limit the changes as much as possible.

This would also allow us to drop the `...Component` suffix from these entities.

```proto
message Component {
  // Component ID, provided by the user on creation.
  string id = 1 [(google.api.field_behavior) = REQUIRED];
  // ...
  // The component configuration.
  oneof component {
    // Start
    Start start = 11;
    // End
    End end = 12;
    // Connector
    Connector connector = 13;
    // Operator
    Operator operator = 14;
    // Iterator
    Iterator iterator = 15;
  }
}

// ...

message ConnectorComponent {
  // Definition name.
  string definition_name = 1 [(google.api.field_behavior) = OPTIONAL];
  // ...
  // Resource name. Leave empty if not created.
  string resource_name = 3 [(google.api.field_behavior) = OPTIONAL];
  // Describes the connector resource details (e.g., configuration to connect with the vendor service).
  ConnectorResource resource = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
  // ...
}

Reply:
@jvallesm's point is good! This indeed create lots of confusion during the adaption of the protobuf and we are struggling with it for quite a long time. I will second replacing the `resource_name` with `connector_name`. The `Resource` is too generic in our platform.

Reply:
> The Connector is the same message used in connector endpoints.

Yep, that's one of the reasons I feared changing this entity would bloat the refactor task.

Removing `resource` is also valid, the point here would be having a clear separation between the pipeline / recipe scope (xxx component) and the VDP scope (connector, even connector/operator definition). I'd add a line in the `Connector`, `Component` and `ComponentDefinition` entities to clarify this.

Reply:
> Removing resource is also valid,

If I remember it correctly, the reason we put resource there is we want to reduce the amount of console requesting backend. So console can only request one `/pipelines/pid` endpoint and get all the data its needs.

Reply:
How about this?

To reduce the scope of the refactor. In this refactor, let's just change to:

```protobuf
  // Connector name. Leave empty if not created.
  string connector_name = 3 [(google.api.field_behavior) = OPTIONAL];
  // Describes the connector details (e.g., configuration to connect with the vendor service).
  Connector connector = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
```

This approach is no doubt an improvement over the original, especially considering that the term "resource" even not appears in Console.

Reply:
I agree to this 👍 

In addition to the provided documentation suggestion (the iterator component is a new concept that probably requires further explanation), I'd change the `Connector` entity documentation to:

```proto
// A connector allows users to query, process or send data to a service or app.
// Users can create and configure connectors that may later be referenced in
// their pipelines via a connector component in order to process their ingested
// unstructured data.
//
// For more information, see
// [Component](https://www.instill.tech/docs/latest/core/concepts/pipeline#pipeline-component)
// in the official documentation.
```

Reply:
It's great. Updated!